src/main/java/edu/ucsb/cs156/happiercows/controllers/CowDeathController.java

```diff
@@ -0,0 +1,118 @@
+
+import edu.ucsb.cs156.happiercows.entities.Commons;
+import edu.ucsb.cs156.happiercows.entities.CommonsPlus;
+import edu.ucsb.cs156.happiercows.entities.User;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.validation.Valid;
+import java.util.Optional;
+import java.time.ZonedDateTime;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.server.ResponseStatusException;
+import org.springframework.http.HttpStatus;
+
+
+@Api(description = "Cow Deaths")
+@RequestMapping("")
+@RestController
+@Slf4j
+public class CowDeathController extends ApiController {
+
+
+    @Autowired
+    CowDeathRepository cowdeathRepository;
+
+    @Autowired
+    CommonsRepository commonsRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+
+
+    @ApiOperation(value = "Creates a new CowDeath entity as Admin")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping("/api/cowdeath")
+    public CowDeath postCowDeath(
+            @ApiParam("commonsId") @RequestParam long commonsId,
+            @ApiParam("userId") @RequestParam long userId,
+            @ApiParam("cowsKilled") @RequestParam Integer cowsKilled,
+            @ApiParam("avgHealth") @RequestParam long avgHealth)
+            throws JsonProcessingException {
+            commonsRepository.findById(commonsId)
+                .orElseThrow(() -> new EntityNotFoundException(Commons.class, commonsId));
+            userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(User.class, userId));
+
+
+        // Ensure that only one can be created. Works but needs approriate tests.
+        //if (cowdeathRepository.getCowsKilledByCommonsIdAndUserId(commonsId, userId).isPresent()){
+        //    throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "CowDeath with inputed commonsId and userId already exists");
+        //}
```

Comment:
Please remove this commented out code.   We actually don't want this feature, and leaving this comment in suggests that perhaps we do. 

In fact, we WILL have the possiblity of multiple cow deaths happening over time for the same commonsId/userId, just on different days.

This is an example of WHY we don't want commented out code in PRs that get merged into main. It creates the potential for confusion among future developers.

Please remove this